src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1962,7 +1969,32 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 diagnostics = BindingDiagnosticBag.Discarded;
             }
 
-            return CreateConversion(expression.Syntax, expression, conversion, isCast: false, conversionGroupOpt: null, targetType, diagnostics);
+            var result = CreateConversion(expression.Syntax, expression, conversion, isCast: false, conversionGroupOpt: null, targetType, diagnostics);
+
+            // A collection expression assigned to a local of ref struct type is treated as local scope.
```

Comment:
>assigned [](http://example.com/codeflow?start=39&length=8)

Should the collection expression be treated as not returnable for all assignments to a local, or just when assigned in a local declaration?

Reply:
It feels like if the local is already returnable we should just quietly permit the collection expr to convert to it. But I'm curious how others feel. Can certainly say if you want that behavior just cast to an array and let it convert back to span..but I'd at least like the diagnostic to lead the user to doing that in that case.

Reply:
We're discussing in teh WG right now :)

Reply:
The logic I mentioned is used to determine the scope of out vars. e.g. `M(stackalloc int[] {...}, out var span)`. And it's determining the widest scope which is valid, not the narrowest.